### PR TITLE
Allow copilot-helper deploy to deploy to a non pipeline referenced environment

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
Co-authored by @WillGibson

### CODE REVIEW

File: dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts [hunk 1]
```diff
@@ -380,6 +384,24 @@ export class TransformedStack extends cdk.Stack {
         });
     }
 
+    private allowPipelineToUseEnvManagerRole() {
+        const pipelineRolePolicy = this.template.getResource("PipelineRolePolicy") as cdk.aws_iam.CfnPolicy;
+        (pipelineRolePolicy.policyDocument.Statement as Array<any>).push({
+            Effect: 'Allow',
+            Action: ['sts:AssumeRole'],
+            Resource: [this.envManagerRole],
+        });
```
Comment (antroy-madetech) [inline dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts:393]: The policy document is the same here as in the next function, so could be pulled out.

File: dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts [hunk 2]
```diff
@@ -432,6 +454,10 @@ export class TransformedStack extends cdk.Stack {
         this.codestarConnection = {arn: codestarConnectionArn, id: codestarConnectionId};
     }
 
+    private loadEnvManagerRole() {
+        this.envManagerRole = `arn:aws:iam::${this.account}:role/${this.appName}-*-EnvManagerRole`;
```
Comment (antroy-madetech) [inline dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts:458]: In fact if the policyDocument were extracted, this won't even be required.

Comment (samuele-mattiuzzo) [reply dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts:458]: @antroy-madetech me and @WillGibson tried a deploy with just the role attached to the PolicyDocument but the build failed so *we think* it needs both for a successful run

Comment (samuele-mattiuzzo) [reply dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts:458]: one thing that can be improved, however is to get rid of the BuildProject's assumeRole blocks for dev and staging as the wildcard we add is going to shadow them

Comment (WillGibson) [reply dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts:458]: I think @ant is getting at...

We could extract the policy document to a method, inline the ``arn:aws:iam::${this.account}:role/${this.appName}-*-EnvManagerRole`` bit, then lose this method.

Comment (samuele-mattiuzzo) [reply dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts]: @WillGibson @antroy-madetech sorted

File: dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts [hunk 3]
```diff
@@ -432,6 +456,8 @@ export class TransformedStack extends cdk.Stack {
         this.codestarConnection = {arn: codestarConnectionArn, id: codestarConnectionId};
     }
 
+
+
```
Comment (antroy-madetech) [inline dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts:460]: Some whitespace snuck past black somehow?

**Review by antroy-madetech — APPROVED** [review]
Just some extraneous looking whitespace, but otherwise LGTM

File: dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts (hunk 3 above)
Comment (samuele-mattiuzzo) [reply dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts]: @antroy-madetech urgh. well, fixed. going to merge it in now and push the same fix to demodjango-deploy

**Review by WillGibson — APPROVED** [review]

**Pull request merged**